### PR TITLE
3871 fix issues introduced by 3604

### PR DESCRIPTION
**Pull request by yuripv**
Pick up the work by @ryao.

### CODE REVIEW

**Comment by prakashsurya** [comment]
@zettabot go

File: usr/src/cmd/zdb/zdb.c [hunk 1]
```diff
@@ -1496,7 +1497,7 @@ dump_full_bpobj(bpobj_t *bpo, char *name, int indent)
 		    (u_longlong_t)bpo->bpo_phys->bpo_subobjs,
 		    bytes, comp, uncomp);
 
-		for (uint64_t i = 0; i < bpo->bpo_phys->bpo_num_subobjs; i++) {
+		for (i = 0; i < bpo->bpo_phys->bpo_num_subobjs; i++) {
```
Comment (ahrens) [inline usr/src/cmd/zdb/zdb.c]: This change seems like a readability regression.  As to consistency with the rest of this file, there are several other instances of this style of declaration, dating back to at least 2008.  Other similar code should join this millennium and adopt c99-style declarations where they improve readability.

File: usr/src/cmd/zdb/zdb.c [hunk 2]
```diff
@@ -1566,7 +1567,8 @@ dump_deadlist(dsl_deadlist_t *dl)
 	    dle = AVL_NEXT(&dl->dl_tree, dle)) {
 		if (dump_opt['d'] >= 5) {
 			char buf[128];
-			(void) snprintf(buf, sizeof (buf), "mintxg %llu -> ",
+			(void) snprintf(buf, sizeof (buf),
+			    "mintxg %llu -> obj %llu",
```
Comment (ahrens) [inline usr/src/cmd/zdb/zdb.c]: good catch, thanks!

File: usr/src/cmd/zdb/zdb.c (hunk 1 above)
Comment (yuripv) [inline usr/src/cmd/zdb/zdb.c]: OK, dropped this change as there are other such style nits already.

**Comment by yuripv** [comment]
Dropped the style change and rebased.

**Comment by ahrens** [comment]
@zettabot go

File: usr/src/cmd/zdb/zdb.c (hunk 2 above)
Comment (pcd1193182) [inline usr/src/cmd/zdb/zdb.c]: I'm amazed there isn't a gcc format warning for this... unless we ignore those in zdb.

**Pull request closed without merging**